include/iterlib/NestIterator.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2015 - present, Facebook
+
+#pragma once
+
+#include "iterlib/Iterator.h"
+#include "iterlib/WrappedIterator.h"
+
+namespace iterlib {
+
+class NestIterator : public WrappedIterator {
```

Comment:
This class name is not very self documenting. 
Can you add block comments explaining the functionality here?